tests/test_accelerator.py

```diff
@@ -31,9 +31,23 @@
 from accelerate.utils import patch_environment
 from accelerate.utils.modeling import get_state_dict_from_offload, load_checkpoint_in_model
 
+class ModelWithTiedWeights(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.linear1 = torch.nn.Linear(2, 4)
+        self.linear2 = torch.nn.Linear(4, 2)
+        self.linear2.weight = self.linear1.weight
+        self.linear2.bias = self.linear1.bias
+
+        # need to add this for compliance with other methods
+        self.weight = self.linear1.weight
+        self.bias = self.linear1.bias
```

Comment:
Do we really need that ? where does it fail ?

Reply:
It used to fail previously. You're right. This part can be safely removed.